services/src/handlers/plots.rs

```diff
@@ -0,0 +1,225 @@
+        .context(error::Operator)?;
+
+    let processor = initialized.query_processor().context(error::Operator)?;
+
+    let query_rect = QueryRectangle {
+        bbox: params.bbox,
+        time_interval: params.time,
+        spatial_resolution: params.spatial_resolution,
+    };
+
+    let query_ctx = ctx.query_context()?;
+
+    Ok(match processor {
+        TypedPlotQueryProcessor::JSON(processor) => {
+            let json = processor
+                .plot_query(query_rect, &query_ctx)
+                .await
+                .context(error::Operator)?;
+
+            Box::new(
+                Response::builder()
+                    .header("Content-Type", "application/json")
+                    .body(json.to_string())
+                    .context(error::HTTP)?,
+            )
+        }
+        TypedPlotQueryProcessor::PNG(processor) => {
+            let png_bytes = processor
+                .plot_query(query_rect, &query_ctx)
+                .await
+                .context(error::Operator)?;
+
+            Box::new(
+                Response::builder()
+                    .header("Content-Type", "application/png")
+                    .body(png_bytes)
+                    .context(error::HTTP)?,
+            )
+        }
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::NaiveDate;
+    use serde_json::json;
+
+    use geoengine_datatypes::primitives::Measurement;
+    use geoengine_datatypes::raster::{Grid2D, RasterDataType, RasterTile2D, TileInformation};
+    use geoengine_datatypes::spatial_reference::SpatialReference;
+    use geoengine_operators::engine::{PlotOperator, RasterOperator, RasterResultDescriptor};
+    use geoengine_operators::mock::{MockRasterSource, MockRasterSourceParams};
+    use geoengine_operators::plot::{Statistics, StatisticsParams};
+
+    use crate::contexts::InMemoryContext;
+    use crate::handlers::handle_rejection;
+    use crate::workflows::workflow::Workflow;
+
+    use super::*;
+    use crate::util::tests::create_session_helper;
+
+    fn example_raster_source() -> Box<dyn RasterOperator> {
+        MockRasterSource {
+            params: MockRasterSourceParams {
+                data: vec![RasterTile2D::new_with_tile_info(
+                    TimeInterval::default(),
+                    TileInformation {
+                        global_geo_transform: Default::default(),
+                        global_tile_position: [0, 0].into(),
+                        tile_size_in_pixels: [3, 2].into(),
+                    },
+                    Grid2D::new([3, 2].into(), vec![1, 2, 3, 4, 5, 6], None).unwrap(),
+                )],
+                result_descriptor: RasterResultDescriptor {
+                    data_type: RasterDataType::U8,
+                    spatial_reference: SpatialReference::epsg_4326().into(),
+                    measurement: Measurement::Unitless,
+                },
+            },
+        }
+        .boxed()
+    }
+
+    #[tokio::test]
+    async fn json() {
```

Comment:
maybe use Lutz's `check_allowed_http_methods` to ensure anything other than GET requests yield a 405

Reply:
I'm not quite sure why you want this, but here you are:

8012f57